cpu/fe310/irq_arch.c

```diff
@@ -108,14 +108,28 @@ void handle_trap(uint32_t mcause)
         }
     }
     else {
+        switch (mcause) {
+        case CAUSE_USER_ECALL:    /* ECALL from user mode */
+        case CAUSE_MACHINE_ECALL: /* ECALL from machine mode */
+        {
+            /* TODO: get the ecall arguments */
+            sched_context_switch_request = 1;
+            /* Increment the return program counter past the ecall
+             * instruction */
+            uint32_t return_pc = read_csr(mepc);
+            write_csr(mepc, return_pc + 4);
+            break;
+        }
+        default:
 #ifdef DEVELHELP
```

Comment:
Would this work with `if (IS_ACTIVE(DEVELHELP)) {` ?

Reply:
Probably, but out of scope here. I'll add that to the list of cleanups for this file